sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/TimerCallback.java

```diff
@@ -25,4 +25,11 @@
 public interface TimerCallback {
   /** Processes the {@link TimerInternals.TimerData TimerData}. */
   void onTimer(TimerInternals.TimerData timer) throws Exception;
+
+  TimerCallback NO_OP = new TimerCallback() {
```

Comment:
It's used in 2 cases:
1) GroupAlsoByWindowsViaOutputBufferDoFn, to simply initialize the processing time and synchronized processing time as part of initializing the TimerInternals. No timers are set at that point.
2) In TriggerTester.advanceInputWatermark(Instant) and TriggerTester.advanceProcessingTime(Instant). Per offline discussion, filed  https://issues.apache.org/jira/browse/BEAM-694 and linked to it from those methods.